department-of-veterans-affairs/vets-website
# Add owned assets

## THIS FORM IS DISABLED IN PRODUCTION

# Income and Asset Statement Form (21P-0969)

## Description
Add a new section with the [Array Builder Pattern(Multiple responses list & loop)](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/patterns/array-builder/README.md)

## Issue
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86366

### Related Issues
n/a

### Epic
[Digitize Income and Asset Statement Form 21P-0969](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/gh/department-of-veterans-affairs/va.gov-team/80925)

### Screenshots
#### Intro page
| Pristine | Required |
| ------- |  ------- | 
|![localhost_3001_income-and-asset-statement-form-21p-0969_ (10)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/67357d7a-320f-41f2-bb69-b07c988801ef)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (11)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/9ea32717-055f-48ce-a3e3-6d581c70d355)

#### Owned asset pages
| First page | Second page |
| ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_ (12)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/9aac6af0-91aa-4386-8dfd-53683b619900)|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (11)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/bf20eb65-c06e-400d-b7f0-7ee4518f5c77)




#### Summary page
| Single Entry | Multiple Entries |
| ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (9)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/58dc0678-0106-4e1a-93e1-4c04ddf67618)|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (10)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/7794d6fd-86ec-42e8-9a57-5bce376b93e0)



#### Review and edit page
| Multiple Entries |
| ------- |
|![reviewandsubmitmultiple](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/ec5dca5e-421e-4bfc-b9c1-10c3b15fa4cd)



## How to run in local environment
1. Check out this branch locally in 'vets-website'
2.  Checkout this [branch](https://github.com/department-of-veterans-affairs/content-build/pull/2075) locally in 'content-build'
3. Run 'vets-api' in your local environment
4. Run the 'vets-website' app with `yarn watch --env entry=static-pages,auth,login-page,profile,dashboard,income-and-asset-statement`

## How to toggle feature flag in local environment
1. Go to [income_and_assets_form_enabled](http://localhost:3000/flipper/features/income_and_assets_form_enabled)

## How to verify
1. Go to `http://localhost:3001/income-and-asset-statement-form-21p-0969` in your browser
2. Sign in with a mock user and 'Start application'
3. Go to the 'Owned assets information' chapter
4. Verify that required fields are required
5. Verify that optional fields are optional
6. Verify that all fields follow validation rules
7. Verify that up to 5 owned assets can be added, edited and deleted
8. Verify that the owned assets are visible and editable on the 'Review and submit' page
9. Verify that the form can be saved via the api endpoint
10. Verify that the form can be submitted

## Testing

## What areas of the site does it impact?
Income and Asset Statement Application

### Quality Assurance & Testing

### Error Handling

### Authentication
